openapi.json

```diff
@@ -0,0 +1,9995 @@
+            "title": "matching_status",
+            "type": "string",
+            "x-nullable": true
+          },
+          "person_id": {
+            "description": "to be documented",
+            "title": "person_id",
+            "type": "integer"
+          },
+          "role": {
+            "description": "to be documented",
+            "enum": [
+              "Commercial Card Holder",
+              "Beneficial Owner",
+              "Company Owner",
+              "Trustee",
+              "Settlor",
+              "Grantor",
+              "Founder",
+              "Beneficiary",
+              "Power of Attorney",
+              "Authorised Signatory",
+              "Purchasing Account",
+              "others"
+            ],
+            "title": "role",
+            "type": "string"
+          },
+          "taggable_entity_id": {
+            "description": "to be documented",
+            "title": "taggable_entity_id",
+            "type": "integer"
+          }
+        },
+        "required": ["person_id", "role", "taggable_entity_id"],
+        "type": "object"
+      },
+      "tag": {
+        "example": {
+          "changed_by": "changed_by",
+          "changed_on": "2000-01-23T04:56:07+00:00",
+          "created_by": "created_by",
+          "created_on": "2000-01-23T04:56:07+00:00",
+          "description": "description",
+          "id": 0,
+          "tagged_entities": [6, 6]
+        },
+        "properties": {
+          "changed_by": {
+            "description": "User name/email of the user who last changed the record via API",
+            "readOnly": true,
+            "title": "changed_by",
+            "type": "string"
+          },
+          "changed_on": {
+            "description": "Timestamp on which the record was last changed, automatically filled by the API",
+            "format": "date-time",
+            "readOnly": true,
+            "title": "changed_on",
+            "type": "string"
+          },
+          "created_by": {
+            "description": "User name/email of the user who inserted the record via API",
+            "readOnly": true,
+            "title": "created_by",
+            "type": "string"
+          },
+          "created_on": {
+            "description": "Timestamp on which the record was created, automatically filled by the API",
+            "format": "date-time",
+            "readOnly": true,
+            "title": "created_on",
+            "type": "string"
+          },
+          "description": {
+            "description": "to be documented",
+            "nullable": true,
+            "title": "description",
+            "type": "string",
+            "x-nullable": true
+          },
+          "id": {
+            "description": "to be documented",
+            "readOnly": true,
+            "title": "id",
+            "type": "integer"
+          },
+          "tagged_entities": { "items": { "type": "integer" }, "type": "array" }
```

Comment:
I'm slightly confused between `tagged_entities`  and `taggable_entity_id` - could the naming be harmonized, either taggable or tagged?

Reply:
Tagged Entities are really the entities which are taggable => Master, Basic, Card & Person

The taggable_entity_ID refers to a single record of one of these taggable entities and the tags framework resolves which exact entity it relates to (so you could think of it like a “taggable entity type” that explains which exact entity the ID / records points to.

@jukietz pls expand if even more confusing.

Reply:
tagged_entities is a relation at tag point to the taggable_entities that have that tag
taggable_entity is an object

The added decription field as above should help you to resolve as
"tagged_entities": { "items": { "type": "integer" }, "type": "array" }
will change to
"tagged_entities": { "items": { "description": "taggable_entity ", "type": "integer" }, "type": "array" }